Competition2018/src/main/java/competition/subsystems/drive/commands/ArcadeDriveWithJoysticksCommand.java

```diff
@@ -30,17 +50,43 @@ public ArcadeDriveWithJoysticksCommand(OperatorInterface oi, DriveSubsystem driv
 
     @Override
     public void initialize() {
-        log.info("Initializing");
+        log.info("Initializing ArcadeDriveWithJoysticksCommand");
         ham.reset();
     }
 
     @Override
     public void execute() {
-        double translate = oi.driverGamepad.getLeftVector().y;
-        double rotate = MathUtils.squareAndRetainSign(oi.driverGamepad.getRightVector().x);
-                
-        double turn = ham.calculateHeadingPower(rotate);
-        driveSubsystem.drive(new XYPair(0, translate), turn);
+        nextTranslateInput = oi.driverGamepad.getLeftVector().y;
+        nextRotateInput = MathUtils.squareAndRetainSign(oi.driverGamepad.getRightVector().x);
+        
+        translateInputDelta = nextTranslateInput - currentTranslateOutput;
+        rotateInputDelta = nextRotateInput - currentRotateOutput;
+        
+        // Rotate
+        if (rotateInputDelta > 0.2 && elevatorSubsystem.getCurrentHeightInInches() > 40) {
+            rotateAdjustedPower += 0.2;
+            turn = ham.calculateHeadingPower(rotateAdjustedPower);
```

Comment:
Can we set the adjusted power in each if statement branch and then have one line at the end which actually runs the final output calculations and sets the powers? i.e., decide what you want to do, and then do it.

Reply:
So basically create the variable inside the if statement?